Out-of-tree build fails

Hi,

Since https://github.com/mbuesch/razer/commit/65f9cdb8b243eff54ca48ebd7f37e822264905dc and the change to run the python setup command in CMAKE_CURRENT_BINARY_DIR rather than CMAKE_CURRENT_SOURCE_DIR, the build breaks with CMake 3.12 (and possibly earlier), which is in Debian unstable/testing. It works fine with CMake 3.9.5 from Debian stable-backports.

I'm 99.9% sure the issue is with CMake as I tried (in a Debian stable chroot) to selectively upgrade packages, and by upgrading CMake the issue starts to occur.

The problem seems to be that while CMake 3.9 correctly copies the 3 pyrazer source files in the build directory, as instructed by the add_custom_command DEPENDS field, CMake 3.12 does not.

CMake 3.9:

```
make -f ui/CMakeFiles/pyrazer.dir/build.make ui/CMakeFiles/pyrazer.dir/depend
make[2]: Entering directory '/home/luca/git/razercfg-debian/build'
cd /home/luca/git/razercfg-debian/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/luca/git/razercfg-debian /home/luca/git/razercfg-debian/ui /home/luca/git/razercfg-debian/build /home/luca/git/razercfg-debian/build/ui /home/luca/git/razercfg-debian/build/ui/CMakeFiles/pyrazer.dir/DependInfo.cmake --color=
Dependee "/home/luca/git/razercfg-debian/build/ui/CMakeFiles/pyrazer.dir/DependInfo.cmake" is newer than depender "/home/luca/git/razercfg-debian/build/ui/CMakeFiles/pyrazer.dir/depend.internal".
Dependee "/home/luca/git/razercfg-debian/build/ui/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/luca/git/razercfg-debian/build/ui/CMakeFiles/pyrazer.dir/depend.internal".
Scanning dependencies of target pyrazer
make[2]: Leaving directory '/home/luca/git/razercfg-debian/build'
make -f ui/CMakeFiles/pyrazer.dir/build.make ui/CMakeFiles/pyrazer.dir/build
make[2]: Entering directory '/home/luca/git/razercfg-debian/build'
[100%] Generating build/stamp
cd /home/luca/git/razercfg-debian/ui && /usr/bin/python3 /home/luca/git/razercfg-debian/build/ui/setup.py build --build-base /home/luca/git/razercfg-debian/build/ui/build
running build
running build_py
creating /home/luca/git/razercfg-debian/build/ui/build
creating /home/luca/git/razercfg-debian/build/ui/build/lib
creating /home/luca/git/razercfg-debian/build/ui/build/lib/pyrazer
copying pyrazer/__init__.py -> /home/luca/git/razercfg-debian/build/ui/build/lib/pyrazer
copying pyrazer/main.py -> /home/luca/git/razercfg-debian/build/ui/build/lib/pyrazer
copying pyrazer/ui.py -> /home/luca/git/razercfg-debian/build/ui/build/lib/pyrazer
running build_scripts
creating /home/luca/git/razercfg-debian/build/ui/build/scripts-3.5
copying and adjusting razercfg -> /home/luca/git/razercfg-debian/build/ui/build/scripts-3.5
copying and adjusting qrazercfg -> /home/luca/git/razercfg-debian/build/ui/build/scripts-3.5
copying and adjusting qrazercfg-applet -> /home/luca/git/razercfg-debian/build/ui/build/scripts-3.5
changing mode of /home/luca/git/razercfg-debian/build/ui/build/scripts-3.5/razercfg from 644 to 755
changing mode of /home/luca/git/razercfg-debian/build/ui/build/scripts-3.5/qrazercfg from 644 to 755
changing mode of /home/luca/git/razercfg-debian/build/ui/build/scripts-3.5/qrazercfg-applet from 644 to 755
cd /home/luca/git/razercfg-debian/ui && /usr/bin/cmake -E touch /home/luca/git/razercfg-debian/build/ui/build/stamp
make[2]: Leaving directory '/home/luca/git/razercfg-debian/build'
[100%] Built target pyrazer
make[1]: Leaving directory '/home/luca/git/razercfg-debian/build'
```

CMake 3.12:

```
make -f ui/CMakeFiles/pyrazer.dir/build.make ui/CMakeFiles/pyrazer.dir/depend
make[2]: Entering directory '/tmp/razer-razercfg-0.40/build'
cd /tmp/razer-razercfg-0.40/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/razer-razercfg-0.40 /tmp/razer-razercfg-0.40/ui /tmp/razer-razercfg-0.40/build /tmp/razer-razercfg-0.40/build/ui /tmp/razer-razercfg-0.40/build/ui/CMakeFiles/pyrazer.dir/DependInfo.cmake --color=
Dependee "/tmp/razer-razercfg-0.40/build/ui/CMakeFiles/pyrazer.dir/DependInfo.cmake" is newer than depender "/tmp/razer-razercfg-0.40/build/ui/CMakeFiles/pyrazer.dir/depend.internal".
Dependee "/tmp/razer-razercfg-0.40/build/ui/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/tmp/razer-razercfg-0.40/build/ui/CMakeFiles/pyrazer.dir/depend.internal".
Scanning dependencies of target pyrazer
make[2]: Leaving directory '/tmp/razer-razercfg-0.40/build'
make -f ui/CMakeFiles/pyrazer.dir/build.make ui/CMakeFiles/pyrazer.dir/build
make[2]: Entering directory '/tmp/razer-razercfg-0.40/build'
[100%] Generating build/stamp
cd /tmp/razer-razercfg-0.40/ui && /usr/bin/python3 /tmp/razer-razercfg-0.40/build/ui/setup.py build --build-base /tmp/razer-razercfg-0.40/build/ui/build
Traceback (most recent call last):
  File "/tmp/razer-razercfg-0.40/build/ui/setup.py", line 4, in <module>
    from pyrazer import RAZER_VERSION
ImportError: No module named 'pyrazer'
ui/CMakeFiles/pyrazer.dir/build.make:62: recipe for target 'ui/build/stamp' failed
make[2]: *** [ui/build/stamp] Error 1
make[2]: Leaving directory '/tmp/razer-razercfg-0.40/build'
CMakeFiles/Makefile2:200: recipe for target 'ui/CMakeFiles/pyrazer.dir/all' failed
make[1]: *** [ui/CMakeFiles/pyrazer.dir/all] Error 2
make[1]: Leaving directory '/tmp/razer-razercfg-0.40/build'
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

$ find ./ui
./ui
./ui/Makefile
./ui/cmake_install.cmake
./ui/CMakeFiles
./ui/CMakeFiles/pyrazer.dir
./ui/CMakeFiles/pyrazer.dir/build.make
./ui/CMakeFiles/pyrazer.dir/DependInfo.cmake
./ui/CMakeFiles/pyrazer.dir/progress.make
./ui/CMakeFiles/pyrazer.dir/cmake_clean.cmake
./ui/CMakeFiles/pyrazer.dir/depend.internal
./ui/CMakeFiles/pyrazer.dir/depend.make
./ui/CMakeFiles/CMakeDirectoryInformation.cmake
./ui/CMakeFiles/progress.marks
./ui/razercfg.desktop
./ui/setup.py
```

Copying the files manually makes it work. I've tried to mess around with the CMake file but I'm no expert so I had no luck unfortunately. I tried to diff the generated makefiles in the ui/ directory between the different CMake versions but I don't see anything obvious that would cause this issue.

Any idea?

Is that a fresh chroot? Could it be that there is some new unexpressed build-dependency?

On a fresh sid chroot, with the tarball downloaded from Github:

```
root@luca-desktop:/tmp/build# cmake ../razer-razercfg-0.40/
-- The C compiler identification is GNU 8.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for libusb.h
-- Looking for libusb.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/build
root@luca-desktop:/tmp/build# make -j1
Scanning dependencies of target razer
[  4%] Building C object librazer/CMakeFiles/razer.dir/buttonmapping.o
[  8%] Building C object librazer/CMakeFiles/razer.dir/profile_emulation.o
[ 12%] Building C object librazer/CMakeFiles/razer.dir/librazer.o
[ 16%] Building C object librazer/CMakeFiles/razer.dir/config.o
[ 20%] Building C object librazer/CMakeFiles/razer.dir/util.o
[ 25%] Building C object librazer/CMakeFiles/razer.dir/synapse.o
[ 29%] Building C object librazer/CMakeFiles/razer.dir/cypress_bootloader.o
[ 33%] Building C object librazer/CMakeFiles/razer.dir/hw_boomslangce.o
[ 37%] Building C object librazer/CMakeFiles/razer.dir/hw_copperhead.o
[ 41%] Building C object librazer/CMakeFiles/razer.dir/hw_deathadder.o
[ 45%] Building C object librazer/CMakeFiles/razer.dir/hw_deathadder2013.o
[ 50%] Building C object librazer/CMakeFiles/razer.dir/hw_deathadder_chroma.o
[ 54%] Building C object librazer/CMakeFiles/razer.dir/hw_krait.o
[ 58%] Building C object librazer/CMakeFiles/razer.dir/hw_lachesis.o
[ 62%] Building C object librazer/CMakeFiles/razer.dir/hw_lachesis5k6.o
[ 66%] Building C object librazer/CMakeFiles/razer.dir/hw_naga.o
[ 70%] Building C object librazer/CMakeFiles/razer.dir/hw_imperator.o
[ 75%] Building C object librazer/CMakeFiles/razer.dir/hw_taipan.o
[ 79%] Building C object librazer/CMakeFiles/razer.dir/hw_mamba_tournament_edition.o
[ 83%] Building C object librazer/CMakeFiles/razer.dir/hw_diamondback_chroma.o
[ 87%] Linking C shared library librazer.so
[ 87%] Built target razer
Scanning dependencies of target razerd
[ 91%] Building C object razerd/CMakeFiles/razerd.dir/razerd.o
[ 95%] Linking C executable razerd
[ 95%] Built target razerd
Scanning dependencies of target pyrazer
[100%] Generating build/stamp
Traceback (most recent call last):
  File "/tmp/build/ui/setup.py", line 4, in <module>
    from pyrazer import RAZER_VERSION
ModuleNotFoundError: No module named 'pyrazer'
make[2]: *** [ui/CMakeFiles/pyrazer.dir/build.make:63: ui/build/stamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:201: ui/CMakeFiles/pyrazer.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
root@luca-desktop:/tmp/build# cmake --version
cmake version 3.12.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

--- a/ui/setup.py.in
+++ b/ui/setup.py.in
@@ -1,5 +1,8 @@
 #!/usr/bin/env python3
 
+import os
+import sys
+sys.path.insert(0, os.getcwd())
 from distutils.core import setup
 from pyrazer import RAZER_VERSION
 

